tests/bluetooth/audio/codec/src/main.c

```diff
@@ -71,3 +72,149 @@ ZTEST(audio_codec_test_suite, test_bt_audio_codec_cfg_get_frame_blocks_per_sdu)
+{
+	const uint8_t metadata[] = {
+		BT_AUDIO_CODEC_DATA(BT_AUDIO_METADATA_TYPE_PREF_CONTEXT,
+				    BT_BYTES_LIST_LE16(BT_AUDIO_CONTEXT_TYPE_UNSPECIFIED |
+						       BT_AUDIO_CONTEXT_TYPE_MEDIA)),
+	};
+	int ret;
+
+	ret = bt_audio_codec_meta_get_pref_context(metadata, ARRAY_SIZE(metadata));
+	zassert_equal(ret, 0x0005, "unexpected return value %d", ret);
+}
+
+ZTEST(audio_codec_test_suite, test_bt_audio_codec_meta_get_stream_context)
+{
+	const uint8_t metadata[] = {
+		BT_AUDIO_CODEC_DATA(BT_AUDIO_METADATA_TYPE_STREAM_CONTEXT,
+				    BT_BYTES_LIST_LE16(BT_AUDIO_CONTEXT_TYPE_UNSPECIFIED |
+						       BT_AUDIO_CONTEXT_TYPE_MEDIA)),
+	};
+	int ret;
+
+	ret = bt_audio_codec_meta_get_stream_context(metadata, ARRAY_SIZE(metadata));
+	zassert_equal(ret, 0x0005, "Unexpected return value %d", ret);
+}
+
+ZTEST(audio_codec_test_suite, test_bt_audio_codec_meta_get_program_info)
+{
+	const uint8_t expected_data[] = {'P', 'r', 'o', 'g', 'r', 'a', 'm', ' ',
+					 'I', 'n', 'f', 'o'};
+	const uint8_t metadata[] = {
+		BT_AUDIO_CODEC_DATA(BT_AUDIO_METADATA_TYPE_PROGRAM_INFO,
+				    'P', 'r', 'o', 'g', 'r', 'a', 'm', ' ', 'I', 'n', 'f', 'o'),
+	};
+	const uint8_t *program_data;
+	int ret;
+
+	ret = bt_audio_codec_meta_get_program_info(metadata, ARRAY_SIZE(metadata), &program_data);
+	zassert_equal(ret, ARRAY_SIZE(expected_data), "Unexpected return value %d", ret);
+	zassert_mem_equal(expected_data, program_data, ARRAY_SIZE(expected_data));
+}
+
+ZTEST(audio_codec_test_suite, test_bt_audio_codec_meta_get_stream_lang)
+{
+	const uint32_t expected_data = sys_get_le24((uint8_t[]){'e', 'n', 'g'});
+	const uint8_t metadata[] = {
+		BT_AUDIO_CODEC_DATA(BT_AUDIO_METADATA_TYPE_STREAM_LANG, 'e', 'n', 'g'),
+	};
+	int ret;
+
+	ret = bt_audio_codec_meta_get_stream_lang(metadata, ARRAY_SIZE(metadata));
+	zassert_equal(ret, expected_data, "Unexpected return value %d", ret);
+}
+
+ZTEST(audio_codec_test_suite, test_bt_audio_codec_meta_get_ccid_list)
+{
+	const uint8_t expected_data[] = {0x05, 0x10, 0x15};
+	const uint8_t metadata[] = {
+		BT_AUDIO_CODEC_DATA(BT_AUDIO_METADATA_TYPE_CCID_LIST, 0x05, 0x10, 0x15),
+	};
+	const uint8_t *ccid_list;
+	int ret;
+
+	ret = bt_audio_codec_meta_get_ccid_list(metadata, ARRAY_SIZE(metadata), &ccid_list);
+	zassert_equal(ret, ARRAY_SIZE(expected_data), "Unexpected return value %d", ret);
+	zassert_mem_equal(expected_data, ccid_list, ARRAY_SIZE(expected_data));
+}
+
+ZTEST(audio_codec_test_suite, test_bt_audio_codec_meta_get_parental_rating)
+{
+	const uint8_t metadata[] = {
+		BT_AUDIO_CODEC_DATA(BT_AUDIO_METADATA_TYPE_PARENTAL_RATING,
+				    BT_AUDIO_PARENTAL_RATING_AGE_10_OR_ABOVE),
+	};
+	int ret;
+
+	ret = bt_audio_codec_meta_get_parental_rating(metadata, ARRAY_SIZE(metadata));
+	zassert_equal(ret, 0x07, "Unexpected return value %d", ret);
+}
+
+ZTEST(audio_codec_test_suite, test_bt_audio_codec_meta_get_program_info_uri)
+{
+	const uint8_t expected_data[] = {'e', 'x', 'a', 'm', 'p', 'l', 'e', '.', 'c', 'o', 'm'};
```

Comment:
Would these normally be 0 terminated UTF-8 strings?

Also, being UTF-8, should there be an extra check on corrupt strings (e.g. cut off in a multi byte char)?

Reply:
These will not be NULL terminated, and will not be truncated. 

The verification of whether the strings are corrupt (or contains invalid UTF-8 characters) are outside the scope of this test and these functions.